tabpy/tabpy_server/handlers/base_handler.py

```diff
@@ -127,12 +127,14 @@ def initialize(self, app):
         self.password = None
         self.eval_timeout = self.settings[SettingsParameters.EvaluateTimeout]
 
+        self.authentication_not_required_error = None # this is fragile, depends on handle_authentication function because this gets set in the handle_authentication function
```

Comment:
Alternatively, you can change `self.not_authorized` to something like `self.auth_error` which can be in one of the states `None`, `NotAuthorized`, and `NotRequired`. With that you won't need to track auth error state with 2 variables.